woocommerce/woocommerce
# Coming soon mode - Match shop page when permalink structure is set to plain

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Slack thread: p1723232801181079-slack-C01SFMVEYAK


This PR fixes an issue with the shop page visibility, where it remains accessible when the site's permalink structure is set to plain and the coming soon mode is enabled with the `restrict to store pages only`.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Navigate to `WooCommerce -> Settings -> Site Visibility` and choose `Coming soon` with `Restrict to store pages only`
3. Navigate to `Settings -> Permalinks` and change `Permalink structure` to `Plain`
4. Open an incognito session and navigate to the frontend.
5. Click `Shop` from the menu.
6. Confirm the coming soon page is rendered.





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Treat post_type=product as a shop page.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
